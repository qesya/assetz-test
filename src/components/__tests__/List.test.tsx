/* eslint-disable testing-library/prefer-screen-queries */
import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import List from '../List';

const exampleData = [
	{
		investorId: 1,
		investmentAccount: 'ABC',
		balance: '10',
		totalBalance: 10,
		totalBalanceDetails: '10'
	},
	{
		investorId: 2,
		investmentAccount: 'DEF',
		balance: '15',
		totalBalance: 15,
		totalBalanceDetails: '15' 
	}
]

describe('<List />', () => {
	test('should has label', () => {
		const { getByText } = render(
			<List
				data={exampleData}
				dataKeys={[
					'investmentAccount',
					'totalBalance',
				]}
			/>
		);
		expect(getByText('investment account')).toBeInTheDocument();
	});

	test('should match snapshot', () => {
		const { asFragment } = render(
			<List
				data={exampleData}
				dataKeys={[
					'investmentAccount',
					'totalBalance',
				]}
			/>
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
