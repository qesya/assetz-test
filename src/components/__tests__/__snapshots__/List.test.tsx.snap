// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ goEvZo"
  >
    <div
      class="sc-gsnTZi frhqIM"
    >
      <div
        class="sc-dkzDqf wUhxQ"
      >
        investment account
      </div>
      <div
        class="sc-dkzDqf wUhxQ"
      >
        total balance
      </div>
    </div>
    <div
      class="sc-gsnTZi kPMvlg"
    >
      <div
        class="sc-dkzDqf wUhxQ"
      >
        ABC
      </div>
      <div
        class="sc-dkzDqf wUhxQ"
      >
        10
      </div>
    </div>
    <div
      class="sc-gsnTZi kPMvlg"
    >
      <div
        class="sc-dkzDqf wUhxQ"
      >
        DEF
      </div>
      <div
        class="sc-dkzDqf wUhxQ"
      >
        15
      </div>
    </div>
  </div>
</DocumentFragment>
`;
